From 48ac7f4431bcd7921571a2287eb31e287f9f8adb Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sun, 26 Aug 2018 15:05:12 +0200
Subject: [PATCH 1/1] efi_loader: pass system table in loaded image protocol

The system table must be passed as a pointer in the loaded image
protocol.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_boottime.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/efi_loader/efi_boottime.c b/lib/efi_loader/efi_boottime.c
index 3935e4f1ce..ef4495ea46 100644
--- a/lib/efi_loader/efi_boottime.c
+++ b/lib/efi_loader/efi_boottime.c
@@ -1493,6 +1493,7 @@ efi_status_t efi_setup_loaded_image(
 
 	info->revision =  EFI_LOADED_IMAGE_PROTOCOL_REVISION;
 	info->file_path = file_path;
+	info->system_table = &systab;
 
 	if (device_path) {
 		info->device_handle = efi_dp_find_obj(device_path, NULL);
-- 
2.18.0

